### PR TITLE
Enable PnP storage stack

### PR DESCRIPTION
**Pull request by Extravert-ir**
Feel free to grab an iso and test

### CODE REVIEW

File: drivers/storage/partmgr/utils.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    _In_ ULONG InputBufferLength,
+    _In_ PVOID OutputBuffer,
+    _In_ ULONG OutputBufferLength,
+    _In_ BOOLEAN InternalDeviceIoControl)
+{
+    PIRP Irp;
+    IO_STATUS_BLOCK IoStatusBlock;
+    PKEVENT Event;
+    NTSTATUS Status;
+    PAGED_CODE();
```
Comment (HBelusca) [inline drivers/storage/partmgr/utils.c:31]:
```suggestion

    PAGED_CODE();
```

File: drivers/storage/partmgr/partmgr.h [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+VOID
+PartMgrReleaseLayoutLock(
+    _In_ PFDO_EXTENSION FDOExtension)
+{
+    PAGED_CODE();
+
+    KeSetEvent(&FDOExtension->SyncEvent, IO_NO_INCREMENT, FALSE);
+}
+
+#endif // _PARTMGR_H_
```
Comment (HBelusca) [inline drivers/storage/partmgr/partmgr.h:170]: Missing trailing newline.

File: drivers/storage/partmgr/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
```
Comment (binarymaster) [inline drivers/storage/partmgr/CMakeLists.txt:1]: `add_subdirectory(partmgr)` should be added to upper level `CMakeLists.txt` file, right? :sweat_smile:

File: drivers/storage/partmgr/partmgr.c [hunk 4]
```diff
@@ -0,0 +1,806 @@
...
+
+    DriverObject->DriverUnload = PartMgrUnload;
+    DriverObject->DriverExtension->AddDevice = PartMgrAddDevice;
+    DriverObject->MajorFunction[IRP_MJ_CREATE]         = ForwardIrpAndForget;
+    DriverObject->MajorFunction[IRP_MJ_CLOSE]          = ForwardIrpAndForget;
+    DriverObject->MajorFunction[IRP_MJ_READ]           = PartMgrReadWrite;
+    DriverObject->MajorFunction[IRP_MJ_WRITE]          = PartMgrReadWrite;
+    DriverObject->MajorFunction[IRP_MJ_DEVICE_CONTROL] = PartMgrDeviceControl;
+    DriverObject->MajorFunction[IRP_MJ_PNP]            = PartMgrPnp;
+
```
Comment (disean) [inline drivers/storage/partmgr/partmgr.c:803]: Power IRPs handler is missing.

**Pull request closed without merging**